internal/artifacts/module_arch_test.go

```diff
@@ -100,20 +105,44 @@ builders:
 		It("Fails on wrong module", func() {
 			Ω(ExecuteSoloBuild(getTestPath("mta"), getResultPath(), nil, "ui5app", os.Getwd)).Should(HaveOccurred())
 		})
+
+		It("Fails on getting default source", func() {
+			Ω(ExecuteSoloBuild(getTestPath("mta"), "", nil, "ui5app", func() (string, error) {
+				return "", errors.New("err")
+			})).Should(HaveOccurred())
+		})
+
+		It("Fails on creation of Location object", func() {
+			counter := 1
+			Ω(ExecuteSoloBuild("", "", nil, "ui5app", func() (string, error) {
+				if counter <= 2 {
```

Comment:
Why is this test waiting for the 3rd call of the function to fail? It seems pretty arbitrary

Reply:
coverage - I wanted to raise an error on dir.Location. The only way of wdGetter. Without this it fails earlier. Yes, it seems ..

Reply:
Isn't there a better way to do this? If any of the logic inside changes (e.g. a call to `wdGetter` is added or removed in this specific flow) the coverage will lessen or the test would fail... also why is it called 3 times in the same flow?

Reply:
as agreed, comment added